daml-lf/transaction-test-lib/src/main/scala/lf/value/test/ValueGenerators.scala

```diff
@@ -565,9 +567,9 @@ object ValueGenerators {
 
   def transactionVersionGen(
       minVersion: TransactionVersion = TransactionVersion.minVersion, // inclusive
-      maxVersion: TransactionVersion = TransactionVersion.maxVersion, // exclusive
+      maxVersion: TransactionVersion = TransactionVersion.maxVersion, // inclusive
```

Comment:
Initially, I tries using `transactionVersionGen` for some test, and was surprised that I couldn't provoke a failure, at a point in time where the failure was expected because the implementation was bugged, but only for `VDev`, and that's when I saw it the upper bound was exclusive.

In the end I moved to using `versionInIncreasingOrder`, which is defined in terms of `TransactionVersion.All` and so covers `VDev` already.

However I think it makes sense to have `transactionVersionGen` also generate tests which cover `VDev`, or else we might not be testing as much as we think.

Reply:
including VDev seems very sensible. I’m slighlty worried that no existing test broke though :confused:

Reply:
The initial idea, was to introduce a range of versions, with `minVersion` being the first version for which the test pass and `mazVersion` the first version for which the test does not pass. 

I think it is a bad idea to include `maxVersion`

In fact the two tests that use `maxVersion` are actually broken, they both cover 1 versions less that what they should.